### PR TITLE
chore(ssa refactor): SSA refactor main

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

This branch just allows us to move faster by removing the mac runner

## Problem\*

Resolves 

## Summary\*

This PR sets out to

### Example

Before:

```

```

After:

```

```

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,9 +19,9 @@ jobs:
           - os: ubuntu
             runner: ubuntu-large
             target: x86_64-linux
-          - os: mac
```
Comment (kevaundray) [inline .github/workflows/test.yml:22]: Not mergable with this change

**Comment by sirasistant** [comment]
Tests are failing due to changes in how SSA handles constants and casts of constants. I added hacky fixes to those in this branch https://github.com/noir-lang/noir/tree/arv/fix_main_hacks but we might want to check if the SSA behavior is correct

**Comment by kevaundray** [comment]
> Tests are failing due to changes in how SSA handles constants and casts of constants. I added hacky fixes to those in this branch https://github.com/noir-lang/noir/tree/arv/fix_main_hacks but we might want to check if the SSA behavior is correct

The SSA behaviour was not correct, but this has been fixed on master. Just merged master in so should be good!

**Review by sirasistant — DISMISSED** [review]

**Comment by kevaundray** [comment]
Currently the brillig oracle test is failing here

**Review by vezenovm — APPROVED** [review]
Good to merge once oracle is fixed

**Pull request merged**